pubsub/gcloud/aio/pubsub/subscriber_client.py

```diff
@@ -1,191 +1,164 @@
 
 from .subscriber_message import SubscriberMessage
 
-
-class FlowControl:
-    def __init__(self, *args: List[Any], **kwargs: Dict[str, Any]) -> None:
+if BUILD_GCLOUD_REST:
+    from requests import Session
+else:
+    from aiohttp import ClientSession as Session  # type: ignore[no-redef]
+
+API_ROOT = 'https://pubsub.googleapis.com'
+VERIFY_SSL = True
+
+SCOPES = [
+    'https://www.googleapis.com/auth/pubsub'
+]
+
+PUBSUB_EMULATOR_HOST = os.environ.get('PUBSUB_EMULATOR_HOST')
+if PUBSUB_EMULATOR_HOST:
+    API_ROOT = f'http://{PUBSUB_EMULATOR_HOST}'
+    VERIFY_SSL = False
+
+
+class SubscriberClient:
+    def __init__(self, *,
+                 service_file: Optional[Union[str, IO[AnyStr]]] = None,
+                 token: Optional[Token] = None,
+                 session: Optional[Session] = None) -> None:
+        self.session = AioSession(session, verify_ssl=VERIFY_SSL)
+        self.token = token or Token(service_file=service_file,
+                                    scopes=SCOPES,
+                                    session=self.session.session)
+
+    async def _headers(self) -> Dict[str, str]:
+        default_headers = {
+            'Content-Type': 'application/json'
+        }
+        if PUBSUB_EMULATOR_HOST:
+            return default_headers
+
+        token = await self.token.get()
+        return {
+            **default_headers,
+            'Authorization': f'Bearer {token}'
+        }
+
+    # https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/create
+    async def create_subscription(self,
+                                  subscription: str,
+                                  topic: str,
+                                  body: Optional[Dict[str, Any]],
+                                  *,
+                                  session: Optional[Session] = None,
+                                  ) -> Dict[str, Any]:
         """
-        FlowControl transitional wrapper.
-        (FlowControl fields docs)[https://github.com/googleapis/python-pubsub/blob/v1.7.0/google/cloud/pubsub_v1/types.py#L124-L166]  # pylint: disable=line-too-long
-        Google uses a named tuple; here are the fields, defaults:
-        - max_bytes: int = 100 * 1024 * 1024
-        - max_messages: int = 1000
-        - max_lease_duration: int = 1 * 60 * 60
-        - max_duration_per_lease_extension: int = 0
+        Create subscription.
         """
-        self._flow_control = _FlowControl(*args, **kwargs)
-
-    def __repr__(self) -> str:
-        result: str = self._flow_control.__repr__()
+        body = {} if not body else body
+        url = f'{API_ROOT}/v1/{subscription}'
+        headers = await self._headers()
+        payload: Dict[str, Any] = {
+            'topic': topic,
+            **body
+        }
+        encoded = json.dumps(payload).encode()
+        s = AioSession(session) if session else self.session
+        resp = await s.put(url, data=encoded, headers=headers)
```

Comment:
still not really familiar with the patterns with sessions, is `raise_for_status` useful here?

Reply:
`raise_for_status` is executed down the stack https://github.com/talkiq/gcloud-aio/blob/master/auth/gcloud/aio/auth/session.py#L115